File: sephora-backend/Application/Dtos/Product/CreateProductDTO.cs
namespace CleanArchitecture.Application.Dtos.Product;

public class CreateProductDto
{
    public string Name { get; set; } = String.Empty;
    public string DescriptionEn { get; set; } = String.Empty;
    public string DescriptionUa { get; set; } = String.Empty;
    public bool Active { get; set; }
    public int BrandId { get; set; }
    public int CategoryId { get; set; }
    public IEnumerable<CreateCharacteristicDto> Characteristics { get; set; } = [];
}

File: sephora-frontend/src/models/product/CreateProductDto.ts
import CreateCharacteristicDto from "../characteristics/CreateCharacteristicDto.ts";

export default interface CreateProductDto {
    get name(): string;

    get descriptionEn(): string;

    get descriptionUa(): string;

    get active(): boolean;

    get brandId(): number;

    get categoryId(): number;

    get characteristics(): CreateCharacteristicDto[];
}
